WalletWasabi/Crypto/ZeroKnowledge/KnowledgeOfDiscreteLog.cs

```diff
@@ -0,0 +1,21 @@
+using NBitcoin.Secp256k1;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public class KnowledgeOfDiscreteLog : KnowledgeOfRepresentation
+	{
+		public KnowledgeOfDiscreteLog(GroupElement nonce, Scalar response)
+			: base(nonce, new[] { response })
```

Comment:
This is called `response` because in the interactive sigma protocol this is what the prover finally responses to the verifier challenge. In our context the word `response` makes no sense at all to me and the real problem is that i don't remember anyone giving a name to it, everybody calls it `s`, `z` and things like that but i think we need a better name. I would prefer `s` as last resource because at least it is not misleading readers.

Reply:
i don't think the name "response" is misleading in any way, the prover challenges themselves using the Fiat-Shamir heuristic, and then responds to that challenge

Reply:
I have no hard feelings about this. I'm fine with both naming conventions as long as we all know what it refers to.

Reply:
Then it stays `response`.